discovery/service.go

```diff
@@ -386,13 +416,28 @@ func (d *Discovery) networkHandler() {
 		// through the (subjectively) new information on their own.
 		case syncReq := <-d.syncRequests:
 			nodePub := syncReq.node.SerializeCompressed()
-			log.Infof("Synchronizing channel graph with %x", nodePub)
-
 			if err := d.synchronize(syncReq); err != nil {
 				log.Errorf("unable to sync graph state with %x: %v",
 					nodePub, err)
 			}
 
+		//...
+		case <-reprocessTimer.C:
+			// Copy the reference in order to be able to nil
+			// announcement slice so that it might be repopulated
+			// during re-processing.
+			copyAnnouncementsRef := d.announcementsToReprocess[:]
```

Comment:
Simply processing the slice backed by `announcementsToReprocess`, then setting the slice to `nil` would result in less lines here. 

Why's the copy necessary?